### PR TITLE
Ops/specializing iet

### PR DESCRIPTION
**Pull request by btrb**
This PR fixes the case in which there isn't affine trees to be converted into OPS.

### CODE REVIEW

File: devito/ops/operator.py [hunk 1]
```diff
@@ -30,10 +30,12 @@ def __init__(self, *args, **kwargs):
     def _specialize_iet(self, iet, **kwargs):
         warning("The OPS backend is still work-in-progress")
 
-        # If there is no iteration tree, then there is no loop to be optimized using OPS.
+        # If there is no iteration trees nor affine trees,
+        # then there is no loop to be optimized using OPS.
         iteration_tree = retrieve_iteration_tree(iet, mode='normal')
-        if not len(iteration_tree):
+        if not len(iteration_tree) or not find_affine_trees(iet):
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: `find_affine_tree` is already called at line 83

you should hoist it at the top of this method and use the result when needed

Comment (btrb) [reply devito/ops/operator.py]: ok, just re-commited considering this comment. thank you

File: devito/ops/operator.py [hunk 2]
```diff
@@ -44,7 +48,7 @@ def _specialize_iet(self, iet, **kwargs):
         # Extract all symbols that need to be converted to ops_dat
         dims = []
         to_dat = set()
-        for section, trees in find_affine_trees(iet).items():
+        for (_, trees) in affine_trees:
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: no need for the parentheses here

File: devito/ops/operator.py [hunk 3]
```diff
@@ -78,7 +82,7 @@ def _specialize_iet(self, iet, **kwargs):
 
         # Generate ops kernels for each offloadable iteration tree
         mapper = {}
-        for n, (section, trees) in enumerate(find_affine_trees(iet).items()):
+        for n, (_, trees) in enumerate(affine_trees):
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: here needed, instead

File: devito/ops/operator.py [hunk 4]
```diff
@@ -30,10 +30,14 @@ def __init__(self, *args, **kwargs):
...
         warning("The OPS backend is still work-in-progress")
 
-        # If there is no iteration tree, then there is no loop to be optimized using OPS.
         iteration_tree = retrieve_iteration_tree(iet, mode='normal')
-        if not len(iteration_tree):
+        affine_trees = find_affine_trees(iet).items()
+
+        # If there is no iteration trees nor affine trees,
+        # then there is no loop to be optimized using OPS.
+        if not len(iteration_tree) or not len(affine_trees):
```
Comment (FabioLuporini) [inline devito/ops/operator.py]: you don't need `len`

Comment (FabioLuporini) [reply devito/ops/operator.py]: in fact, I think this condition should just be

```
if not affine_trees: 
     ....
```

File: devito/ops/operator.py (hunk 2 above)
Comment (btrb) [reply devito/ops/operator.py]: ok

File: devito/ops/operator.py (hunk 4 above)
Comment (btrb) [reply devito/ops/operator.py]: ok, fixing

**Comment by btrb** [comment]
closing in favor of [#953](https://github.com/opesci/devito/pull/953)

**Pull request closed without merging**